lib/game.rb

```diff
@@ -0,0 +1,101 @@
+require 'pry'
+
+class Game
+
+attr_accessor :player_1, :player_2, :board, :the_winner
+
+  WIN_COMBINATIONS = [
+    [0,1,2], # Top row
+    [3,4,5], # Middle row
+    [6,7,8], # Bottom row
+    [0,3,6], # Left
+    [1,4,7], # Middle
+    [2,5,8], #Right
+    [0,4,8], #L Diag
+    [2,4,6]  #R Diag
+  ]
+
+  def initialize(player_1=Players::Human.new("X"), player_2=Players::Human.new("O"), board=Board.new)
+      @player_1 = player_1
+      @player_2 = player_2
+      @board = board
+      @the_winner=""
+  end
+
+  def current_player
+    if @board.turn_count % 2 == 0
+      @player_1
+    else
+      @player_2
+    end
+  end
+
+  def won?
+    winning_combo = nil
+    WIN_COMBINATIONS.each do |combo|
+      if combo.all? do |pos|
+        @board.position(pos+1) == "X"
+        end
+        winning_combo = combo
+      elsif combo.all? do |pos|
+          @board.position(pos+1) == "O"
+        end
+        winning_combo = combo
+      else
+        next
+      end #end if
+    end #Match all winning combos to the current board
+    winning_combo
+  end
+
+  def draw?
+    if @board.full? && won? == nil
+      true
+    else
+      false
+    end
+  end
+
+  def over?
+    if won? || draw?
+      true
+    else
+      false
+    end
+  end
+
+  def winner
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if winning_combo = won?
  board.cells[winning_combo.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.